test/jdk/javax/swing/AbstractButton/5049549/bug5049549.java

```diff
@@ -39,18 +35,40 @@
 import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
 
+/* @test
+   @bug 5049549 7132413
+   @summary Tests that the proper icon is used for different states.
+   @library ../../regtesthelpers
+   @build Blocker
+   @run main/manual bug5049549
+*/
+
 public class bug5049549 {
 
-    private static ImageIcon DE = new ImageIcon(bug5049549.class.getResource("DE1.gif"));
-    private static ImageIcon DI = new ImageIcon(bug5049549.class.getResource("DI1.gif"));
-    private static ImageIcon DS = new ImageIcon(bug5049549.class.getResource("DS1.gif"));
-    private static ImageIcon RO = new ImageIcon(bug5049549.class.getResource("RO1.gif"));
-    private static ImageIcon RS = new ImageIcon(bug5049549.class.getResource("RS1.gif"));
-    private static ImageIcon SE = new ImageIcon(bug5049549.class.getResource("SE1.gif"));
-    private static ImageIcon PR = new ImageIcon(bug5049549.class.getResource("PR1.gif"));
+    private static Icon DE = generateImage("DE");
+    private static Icon DI = generateImage("DI");
+    private static Icon DS = generateImage("DS");
+    private static Icon RO = generateImage("RO");
+    private static Icon RS = generateImage("RS");
+    private static Icon SE = generateImage("SE");
+    private static Icon PR = generateImage("PR");
```

Comment:
I suggest marking all of them `final`:
```suggestion
    private static final Icon DE = generateImage("DE");
    private static final Icon DI = generateImage("DI");
    private static final Icon DS = generateImage("DS");
    private static final Icon RO = generateImage("RO");
    private static final Icon RS = generateImage("RS");
    private static final Icon SE = generateImage("SE");
    private static final Icon PR = generateImage("PR");
```